helix-core/src/test/java/org/apache/helix/ZkTestHelper.java

```diff
@@ -146,6 +146,70 @@ public void process(WatchedEvent event) {
     LOG.info("After expiry. sessionId: " + Long.toHexString(curZookeeper.getSessionId()));
   }
 
+  public static void expireSharedZkClientSession(HelixZkClient client) throws Exception {
+    final CountDownLatch waitExpireSession = new CountDownLatch(1);
+    final ZkClient zkClient = (ZkClient) client;
+
+    IZkStateListener listener = new IZkStateListener() {
+      @Override
+      public void handleStateChanged(KeeperState state) throws Exception {
+        LOG.info("IZkStateListener#handleStateChanged, state: " + state);
+        if (state == KeeperState.Expired) {
+          waitExpireSession.countDown();
```

Comment:
I guess this wait only check if there is an expiring happened. The original wait is to ensure the session has been re-established. So if you want to achieve the same goal, maybe we need to compare the older session and newer session then end the wait.
Given that saying, I guess you can just modify the existing method to include this check. Note that if this not happen, we can directly fail the test by throwing an exception.

Reply:
You are right, here this latch guarantees expiring happens. But later, the code still ensures new session established.

`     waitExpireSession.await();
    zkClient.unsubscribeStateChanges(listener);

    connection = (ZkConnection) zkClient.getConnection();
    curZookeeper = connection.getZookeeper();

    // wait util connected
    while (curZookeeper.getState() != States.CONNECTED) {
      Thread.sleep(10);
    } `

Reply:
This work. Since it should work for all the client, can we just modify the original expiring method? 2 similar methods for the same purpose is not good.

Reply:
See my comment later about how to enhance with handleStateChanged. If we agree with that approach. I will use this logic in TestHelper.expireSession() and remove this one.